server/src-lib/Hasura/GraphQL/Transport/WebSocket/Protocol.hs

```diff
@@ -22,6 +22,8 @@ import           Hasura.EncJSON
 import           Hasura.GraphQL.Transport.HTTP.Protocol
 import           Hasura.Prelude
 
+-- TODO what is the provenance? It looks like this comes from the client.
+--      Can any client who can guess another client's OperationId do anything naughty?
```

Comment:
Nope, an `OperationId` is unique per websocket.

Reply:
Thanks. Could you elaborate? Do you mean the operationId + someotherkey together form a unique key so we don't have to worry about operationId collisions?

Reply:
I'm not sure I follow what you are asking, the type represents `id` as documented here: https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md.